src/lib/jsonParseV2/parseRegistration.cpp

```diff
@@ -0,0 +1,355 @@
+    else
+    {
+      *errorStringP = "invalid value of /supportedForwardingMode/";
+      return false;
+    }
+  }
+  else
+  {
+    providerP->supportedForwardingMode = ngsiv2::ForwardAll;
+  }
+
+  if (provider.HasMember("legacyForwarding"))
+  {
+    const rapidjson::Value& legacyForwarding = provider["legacyForwarding"];
+
+    if (!legacyForwarding.IsBool())
+    {
+      *errorStringP = "the field /legacyForwarding/ must be a boolean and set to /true/";
+      return false;
+    }
+
+    bool legacyForwardingValue = legacyForwarding.GetBool();
+    if (legacyForwardingValue != true)
+    {
+      *errorStringP = "the field /legacyForwarding/ must be set to /true/";
+      return false;
+    }
+  }
+  else
+  {
+    *errorStringP = "the field /legacyForwarding/ must be present as a boolean and set to /true/";
+    return false;
+  }
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* parseRegistration -
+*/
+std::string parseRegistration(ConnectionInfo* ciP, ngsiv2::Registration* regP)
+{
+  std::string          errorString;
+  rapidjson::Document  document;
+
+  document.Parse(ciP->payload);
+
+  if (document.HasParseError())
+  {
+    return badInput(ciP, ERROR_DESC_PARSE);
+  }
+
+  if (!document.IsObject())
+  {
+    return badInput(ciP, ERROR_DESC_BAD_REQUEST_NOT_A_JSON_OBJECT);
+  }
+
+  if (document.ObjectEmpty())
+  {
+    return badInput(ciP, ERROR_DESC_BAD_REQUEST_EMPTY_PAYLOAD);
+  }
+
+
+  //
+  // Extracting field "description"
+  //
+  Opt<std::string> description = getStringOpt(document, "description");
+
+  if (!description.ok())
+  {
+    return badInput(ciP, "invalid description");
+  }
+  else if (description.given)
+  {
+    regP->description = description.value;
+    if (regP->description.length() > MAX_DESCRIPTION_LENGTH)
+    {
+      return badInput(ciP, "max description length exceeded");
+    }
+
+    if (forbiddenChars(regP->description.c_str()))
+    {
+      return badInput(ciP, "forbidden characters in description");
+    }
+  }
+
+
+  //
+  // Extracting field "dataProvided"
+  //
+  if (!document.HasMember("dataProvided"))
+  {
+    return badInput(ciP, "the field /dataProvided/ is missing in payload");
+  }
+
+  const rapidjson::Value& dataProvided = document["dataProvided"];
+
+  if (dataProvidedParse(ciP, &regP->dataProvided, dataProvided, &errorString) == false)
+  {
+    return badInput(ciP, errorString);
+  }
+
+
+  //
+  // Extracting field "provider"
+  //
+  if (!document.HasMember("provider"))
+  {
+    return badInput(ciP, "the field /provider/ is missing in payload");
+  }
+
+  const rapidjson::Value& provider = document["provider"];
+
+  if (providerParse(ciP, &regP->provider, provider, &errorString) == false)
+  {
+    return badInput(ciP, errorString);
+  }
+
+
+  //
+  // Extracting field "expires"
+  //
+  if (document.HasMember("expires"))
```

Comment:
I [suggested an alternative for this block](https://github.com/apinf/fiware-orion/pull/14#discussion_r161169060) that as far as I undertand have two advantages over the current alternative:

* Checks expires emptyness (meaning permanent)
* Check ISO8601 format

And a more importan advantage: makes the way of parsing expiration in suscriptions and registrations uniform.